systemtest/src/test/java/io/strimzi/systemtest/KafkaClusterTest.java

```diff
@@ -213,4 +215,30 @@ private void waitForZkMntr(String pod, Pattern pattern) {
         });
     }
 
+    @Test
+    @KafkaCluster(name = "my-cluster",
+            cmConfiguration = {@KafkaCluster.CmConfiguration(key = "zookeeper-healthcheck-delay", value = "30"),
+                    @KafkaCluster.CmConfiguration(key = "zookeeper-healthcheck-timeout", value = "10"),
+                    @KafkaCluster.CmConfiguration(key = "kafka-healthcheck-delay", value = "30"),
+                    @KafkaCluster.CmConfiguration(key = "kafka-healthcheck-timeout", value = "10"),
+                    @KafkaCluster.CmConfiguration(key = "KAFKA_DEFAULT_REPLICATION_FACTOR", value = "2"),
+                    @KafkaCluster.CmConfiguration(key = "KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR", value = "5"),
+                    @KafkaCluster.CmConfiguration(key = "KAFKA_TRANSACTION_STATE_LOG_REPLICATION_FACTOR", value = "5")
+            })
+    public void testClusterWithCustomParameters() {
+        // kafka cluster already deployed via annotation
+        String clusterName = "my-cluster";
+        LOGGER.info("Running clusterWithCustomParameters with cluster {}", clusterName);
+
+        String jsonString = kubeClient.get("cm", clusterName);
+
+        assertThat(jsonString, valueOfCmEqualsTo("zookeeper-healthcheck-delay", "30"));
```

Comment:
Personally I don't see much benefit of using the `valueOfCmEqualsTo` matcher like this: You end up parsing the `jsonString` for each assertion. If you just parsed it once and got the data `Map<String, String>` you can then just as easily assert against the keys of that using regular Junit rather than using Hamcrest. But it's up to you.

Reply:
I think that using this matcher and Hamcrest we can avoid code duplication in future when we will have more automation tests and when we need to check CM. Another advantage is: if we have changes in CM structure (for example, JSON key "data" will be renamed to "config") we can make a correction only in one place. But I would like to discuss with you this solution one more time.